cmd/juju/controller/addmodel_test.go

```diff
@@ -209,6 +231,47 @@ func (s *AddModelSuite) TestCredentialsNoDefaultCloud(c *gc.C) {
     juju add-model \[flags\] \<model-name\> cloud\[/region\]`)
 }
 
+func (s *AddModelSuite) TestCredentialsOneCached(c *gc.C) {
+	// Disable empty auth and clear the local credentials,
+	// forcing a check for credentials in the controller.
+	s.PatchValue(&s.fakeCloudAPI.authTypes, []cloud.AuthType{cloud.AccessKeyAuthType})
+	delete(s.store.Credentials, "aws")
+
+	// Cache just a single credential in the controller,
+	// so it is selected automatically.
+	credentialTag := names.NewCloudCredentialTag("aws/foo/secrets")
+	s.PatchValue(&s.fakeCloudAPI.credentials, []names.CloudCredentialTag{credentialTag})
+
+	_, err := s.run(c, "test")
+	c.Assert(err, jc.ErrorIsNil)
+
+	c.Assert(s.fakeAddModelAPI.cloudCredential, gc.Equals, credentialTag)
+}
+
+func (s *AddModelSuite) TestCredentialsDetected(c *gc.C) {
+	// Disable empty auth and clear the local credentials,
+	// forcing a check for credentials in the controller.
+	// There are multiple credentials in the controller,
+	// so none of them will be chosen by default.
+	s.PatchValue(&s.fakeCloudAPI.authTypes, []cloud.AuthType{cloud.AccessKeyAuthType})
+
+	// Delete all local credentials, so we don't choose
+	// any of them to upload. This will force credential
+	// detection.
+	delete(s.store.Credentials, "aws")
+
+	_, err := s.run(c, "test")
+	c.Assert(err, jc.ErrorIsNil)
```

Comment:
Should this be some sort of ambiguous credential error if there are multiple in the controller?